src/Fantomas.Core/Trivia.fs

```diff
@@ -207,16 +213,32 @@ let private addTriviaToTriviaNode (startOfSourceCode: int) (triviaNodes: TriviaN
 
     // Newlines are only relevant if they occur after the first line of source code
     | { Item = Newline; Range = range } when (range.StartLine > startOfSourceCode) ->
+        let canAddNewline triviaContent =
+            let numberOfNewlinesAtEndOfContent triviaContent =
```

Comment:
I wonder if we could not collect the newlines in the first place in `collectTriviaFromBlankLines` if we know there is a limit.

If we detected a newline at lines `[ 2; 3; 5]` and the limit is 1, we could potentially only create two Newline trivia `[ Newline at 3; Newline at 5 ]`.
I'm not sure what would be more performant.